--- a/src/win32_window.c
+++ b/src/win32_window.c
@@ -1146,11 +1146,6 @@
             break;
         }

-        case WM_ERASEBKGND:
-        {
-            return TRUE;
-        }
-
         case WM_NCACTIVATE:
         case WM_NCPAINT:
         {
@@ -1437,6 +1432,7 @@
     wc.hInstance     = _glfw.win32.instance;
     wc.hCursor       = LoadCursorW(NULL, IDC_ARROW);
     wc.lpszClassName = _GLFW_WNDCLASSNAME;
+    wc.hbrBackground = (HBRUSH)GetStockObject(BLACK_BRUSH);

     // Load user-provided icon if available
     wc.hIcon = LoadImageW(GetModuleHandleW(NULL),
